packages/ra-ui-materialui/src/list/filter/FilterButton.tsx

```diff
@@ -73,6 +73,10 @@ const FilterButton = (props: FilterButtonProps): JSX.Element => {
         [showFilter, setOpen]
     );
 
+    if (!hiddenFilters) {
```

Comment:
this will throw as soon as all the filters are shown. Not good!

Reply:
When all the filters are shown, `hiddenFilters` will be an empty array, `!hiddenFilters` will be false.
I can improve it by asking for `undefined`, but It should work as it is

Reply:
Yes, you're right. But see my (edited) comment: your code is correct but probably not the clearest way to do it.

Reply:
You are right, I didn't ask if `filters` was defined, because I was under the wrong idea that I was returning, which I cannot do because of the hooks.